design/ceph/object/store.md

```diff
@@ -712,3 +712,64 @@ For external CephCephObjectStores (i.e., when `spec.gateway.externalRgwEndpoints
 vhost-style addressing should be configured on the host cluster, and `hosting.dnsNames` is
 irrelevant. The default `advertiseEndpoint` for external CephObjectStores is the first entry in the
 `spec.gateway.externalRgwEndpoints` list, which users should be able to override if desired.
+
+### Advanced user configuration
+
+RGW is a very active project with many existing features and regular new feature additions. Rook
+maintainers have a hard time catching up with RGW's latest features. In an effort to help users who
+want access to rare and new features, Rook can allow users to set arbitrary Ceph configs in their
+CephObjectStore CR. RGW config reference: https://docs.ceph.com/en/reef/radosgw/config-ref/
+
+This is an advanced feature. It allows users to modify many untested aspects of the RGW and object
+store. Any user configs that break the object store should be a responsibility of the user.
+
+```yaml
+spec:
+  metadataPool: # ...
+  dataPool: # ...
+  gateway: # ...
+
+  cephConfig:
+    rgw_enable_apis: s3, s3website, swift, swift_auth, sts, iam, notifications # e.g. 'admin' removed
+    rgw_gc_max_concurrent_io: "4" # e.g, reduce garbage collection IO
+    debug_rgw: "20"
+```
+
+CephCluster has a similar configuration section called `config`. `config` may be somewhat too vague
+in the context of the CephObjectStore CR. `cephConfig` is a clearer naming for users and give
+context that Ceph configs are expected.
```

Comment:
I also considered `advancedCephConfig` but figured "advanced" might end up being unnecessary clutter.

Reply:
I agree on `advanced` being unnecessary as a prefix.  I also like the field name being consistent with cephCluster.

Reply:
CephCluster just uses `config`. I started the design using that, but I had doubts about it. In CephCluster, `config` can be assumed to be a Ceph config, but in context of the ObjectStore, it seemed a little vague to me. I changed to `cephConfig` in this context, but I do want other opinions on this.

Reply:
We have a `gateway` config section, which represents the "daemon" so I'm a bit reluctant to use `daemon` or `rgw` in the config name. I also see benefit in your suggestions.

In reality, these configs do only affect the RGW daemon itself. They aren't configs that are for the whole "object store" in the way zone configs affect the store itself. In reflection, putting these settings under `gateway` might be better. @travisn thoughts?

I'm still trying to figure out good wording for an overall config section.

Something like this

```yaml
spec:
  gateway:
    # ... 
    cephConfigs:
      startup:  # <-- possibly easier to understand than "initialization"?
        flag_name: flag_value
      runtime:
        conf_name: conf_value
```

Reply:
`parameters` seems a little vague to me. I also think a lot of users might take the wrong meaning from `monStore`. Ceph mons do provide a mon store for arbitrary kv data, which is separate from the centralized config store.

If we aren't going to include a `cephConfig`/`rgwConfig` heading to add context, I feel like `rgwCommandFlags` and `rgwConfigs` is clearest for users.

Reply:
As far as applying flags, I did a little experimentation to make sure Ceph handles it well. It looks like the last flag in the list sets the value for `radosgw`, which is good. We can append flag parameters to the end of the CLI command, and what the user sets should get precedence.

```sh
[root@e37cb82f0329 /]# radosgw -c /tmp/rgw.conf
global_init: unable to open config file from search list /tmp/rgw.conf
[root@e37cb82f0329 /]# radosgw -c /tmp/rgw.conf -c /tmp/rgw2.conf
global_init: unable to open config file from search list /tmp/rgw2.conf
```

Reply:
The `rgwConfigs` and `rgwCommandFlags` look good to me